rfcs/45-stageless.md

```diff
@@ -0,0 +1,564 @@
+
+#### Startup systems
+
+Startup systems are stored in their own schedule, with the `CoreSchedule::Startup` label.
+When using the runner added by `MinimalPlugins` and `DefaultPlugins`, this schedule will run exactly once on app startup.
+
+You can add startup systems with the `.add_startup_system(my_system)` method on `App`, which is simply sugar for `.add_system(my_system.to_schedule(CoreSchedule::Startup))`.
+Startup systems can be run again by running this schedule from within an exclusive system.
+
+### Introduction to system configuration
+
+When a function is added to a schedule, it is turned into a `RawSystem` and combined with a `SystemConfig` struct to create a `ConfiguredSystem`.
+
+A system may be configured in the following ways:
+
+- it may have any number of **labels** (which implement the `SystemLabel` trait)
+  - labels are used by other systems to determine ordering constraints
+  - e.g. `.add_system(gravity.label(GameLogic::Physics))`
+  - labels themselves may be configured, allowing you to define high-level structure in your `App` in a single place
+    - e.g `app.configure_label(GameLogic::Physics, SystemConfig::new().label(GameLogic::Physics).after(GameLogic::Input))`
+- it may have ordering constraints, causing it to run before or after other systems
+  - there are several variations on this, see the next section for details
+  - e.g. `.add_system(player_controls.before(GameLogic::Physics))`
+- it may have one or more **run criteria** attached
+  - a system is only executed if all of its run criteria returns `true`
+  - **states** are a special, more complex pattern that use run criteria to determine whether or not a system should run in a current state
+  - e.g. `.add_system(physics.run_if(GameLogic::Physics))`
+
+System configuration can be stored in a `SystemConfig` struct. This can be useful to reuse, compose and quickly apply complex configuration strategies, before applying these strategies to labels or individual systems.
+
+If you just want to run your game logic systems in the middle of your schedule, after input is processed but before rendering occurs, add the premade `CoreLabel::GameLogic` to them.
+
+### Label configuration
+
+Each system label has its own associated `SystemConfig`, stored in the corresponding `Schedule`.
+This configuration is applied in an additive fashion to each system with that label.
+
+You can define labels in a standalone fashion, configuring them at the time of creation:
+
+```rust
+#[derive(SystemLabel)]
+enum Physics {
+    Forces,
+    CollisionDetection,
+    CollisionHandling,
+}
+
+impl Plugin for PhysicsPlugin{
+
+    fn build(app: &mut App){
+        // Within each frame, physics logic needs to occur after input handling, but before rendering
+        let mut common_physics_config = SystemConfig::new().after(CoreLabel::InputHandling).before(CoreLabel::Rendering);
+

```

Comment:
```suggestion

use Physics::Forces;
```
to make the rest much more likely to compile

Reply:
Fixed.